src/dezerialize.ts

```diff
@@ -32,6 +33,12 @@ import {
 } from "./types";
 import { ZodTypes } from "./zod-types";
 
+type DezerializerOptions = {
+  superRefinements?: { [key: string]: (value: any, ctx: any) => void };
+  transforms?: { [key: string]: (value: any, ctx: any) => void };
+  preprocesses?: { [key: string]: (value: any, ctx: any) => void };
```

Comment:
could we type `value` as `unknown` and `ctx` as `RefinementCtx` (exported by zod) here? I think the return type for all off those could also be `Promise<void>`

Reply:
1. Re: `ctx` as `RefinementCtx`, sure.
2. Re: `value` as `unknown`, pardon my great ignorance of TS here, but won't that put a burden on users to cast their functions' arguments in the function body (e.g., if they want to enforce a more specific type like `Date`)?
3. Re: return type, good catch. Those could actually be returning other values directly, not even including async `transform`s (the docs don't mention async `preprocess`es or `superRefine`s). (Not sure why TS didn't catch the problem with `void` when the latter two returned various values in test code.) Depending on your answers above, so far I've got now:

```js
{
  superRefinements?: { [key: string]: (value: any, ctx: z.RefinementCtx) => void };
  transforms?: { [key: string]: (value: any, ctx: z.RefinementCtx) => unknown };
  preprocesses?: { [key: string]: (value: any, ctx: z.RefinementCtx) => unknown };
};
```

Reply:
Sorry, just updated my data and summary above.

Reply:
2. It does put a burden on that side, but I think it's the right kind of burden. Since `zod`'s primary use case (as I see it) is about making blobs of potentially unknown data safe to use by parsing it, it seems right to me to use `unknown` here. Users can still opt out of that by coercing the value into whatever type they think they can safely assume there.
That said, the best type would ofc be the actual type of the thing where the refinement is used, but I suppose there is no feasible way of using generic types to accomplish that?!
3. Makes sense, good catch!